### ISSUE TITLE
CA1707 should not be raised for conversion operators

### ISSUE THREAD

**Issue by dpaoliello** [dotnet/roslyn-analyzers#1288]
#### Analyzer package

Microsoft.CodeQuality.Analyzers

#### Analyzer

IdentifiersShouldNotContainUnderscores

#### Repro steps

```csharp
public class Class1
{
    public static implicit operator Class1(int value) => new Class1();
}
```

#### Expected behavior

CA1707 is not raised (this is a generated name)

#### Actual behavior

`warning CA1707: Remove the underscores from member name Class1.implicit operator Class1(int)`

**Comment by dpaoliello** [comment]
Looks like the fix is to extend #970 by also checking for `MethodKind.Conversion`

**Comment by mavasani** [comment]
Fixed with https://github.com/dotnet/roslyn-analyzers/pull/1329
